url9d32402837_dusan_rychnovsky_chess_endgames_tgz-pJ8-cz_dusanrychnovsky_chessendgames_core_strategies_PrecomputedValuesJ8/src/cz/dusanrychnovsky/chessendgames/core/Move.java

```diff
@@ -0,0 +1,144 @@
+package cz.dusanrychnovsky.chessendgames.core;
+
+import java.io.Serializable;
+import java.util.List;
+
+/**
+ * 
+ * @author Dušan Rychnovský
+ *
+ */
+public class Move implements Serializable
+{
+	private static final long serialVersionUID = 1L;
+	
+	private final Piece piece;
+	private final Range range;
+	private final Position from;
+	private final Position to;
+	
+	/**
+	 * 
+	 * @param piece
+	 * @param from
+	 * @param to
+	 */
+	public Move(Piece piece, Position from, Position to) {
+		this.piece = piece;
+		this.range = Range.get(from, to);
+		this.from = from;
+		this.to = to;
+	}
+	
+	/**
+	 * 
+	 * @return
+	 */
+	public Piece getPiece() {
+		return piece;
+	}
+		
+	/**
+	 * 
+	 * @return
+	 */
+	public Position getTo() {
+		return to;
+	}
+	
+	/**
+	 * Returns true if and only if the represented move is valid with regards
+	 * to the given situation.
+	 * 
+	 * @param situation
+	 * @return
+	 */
+	public boolean isValid(Situation situation)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 71-80, lines 101-115 into separate methods.